pandas/core/indexes/base.py

```diff
@@ -1832,15 +1832,31 @@ def __contains__(self, key):
             return False
 
     _index_shared_docs['contains'] = """
-        return a boolean if this key is IN the index
+        Return a boolean if key is in the index.
+
+        This function returns True if key is in the index and
+        returns False if key is not in index.
 
         Parameters
         ----------
-        key : object
+        key : immutable object
+            Key to be searched.
 
         Returns
         -------
-        boolean
+        bool
+            Result of the key search.
+
+        Notes
+        -----
+            CategoricalIndex is a sub-class of Index.
+
+        Examples
+        --------
+        >>> pd.CategoricalIndex([2000, 2001, 2012]).contains(2001)
```

Comment:
Best to just use a regular Index here, not a CategoricalIndex.

Reply:
pd.Index.contains and pd.CategoricalIndex.contains have shared documentation 
and a CategoricalIndex example is valid for Index but not the other way around since CategoricalIndex is a sub-class of Index

Reply:
yeah for this particular case I would separate the doc-strings, IOW 1 more Index and a separate one for CategoricalIndex. Can use a shared doc if that works (with templates)